controller/src/main/java/org/jboss/as/controller/AbstractAttributeDefinitionBuilder.java

```diff
@@ -334,6 +334,51 @@ public BUILDER addAlternatives(String... alternatives) {
     }
 
     /**
+     * Removes {@link AttributeDefinition#getAlternatives() names of alternative attributes} from the set of those that should not
+     * be defined if this attribute is defined.
+     * @param alternatives the attribute names
+     * @return a builder that can be used to continue building the attribute definition
+     */
+    public BUILDER removeAlternatives(String... alternatives) {
+        if (this.alternatives == null) {
+            return (BUILDER) this;
+        } else {
+            for (String alternative : alternatives) {
+                if (isAlternativePresent(alternative)) {
+                    int length = this.alternatives.length;
+                    String[] newAlternatives = new String[length - 1];
+                    int k = 0;
+                    for (String alt : this.alternatives) {
+                        if (alt != alternative) {
+                            newAlternatives[k] = alt;
+                            k++;
+                        }
+                    }
+                    this.alternatives = newAlternatives;
+                }
+            }
+        }
+        return (BUILDER) this;
+    }
+
+    /**
+     * Checks if an alternative has been recorded in {@link AttributeDefinition#getAlternatives() names of alternative attributes}
+     * @param alternative the alternative
+     * @return a builder that can be used to continue building the attribute definition
+     */
+    protected boolean isAlternativePresent(final String alternative) {
```

Comment:
This should be private.  The similar isFlagPresent method should have been private too, but since it was released as protected I don't want to reduce the visibility. But this one is new so let's lock it down.